ui/src/app/views/concept-table/component.tsx

```diff
@@ -0,0 +1,140 @@
+  loading: boolean;
+  onSelectedChanged: Function;
+}
+
+interface ConceptTableState {
+  selectedConcepts: Concept[];
+  selectedVocabularies: string[];
+}
+
+export class ConceptTable extends React.Component<ConceptTableProps, ConceptTableState> {
+
+  private dt: DataTable;
+  private filterImageSrc: string;
+
+  constructor(props: ConceptTableProps) {
+    super(props);
+    this.state = {
+      selectedConcepts: [],
+      selectedVocabularies: []
+    };
+    this.filterImageSrc = 'filter';
+  }
+
+  conceptSynonymColTemplate(rowData) {
+    return <div>{fp.uniq(rowData.conceptSynonyms).join(', ')}</div>;
+  }
+
+  updateSelectedConceptList(selectedConcepts) {
+    this.setState({selectedConcepts : selectedConcepts});
+    this.props.onSelectedChanged(selectedConcepts);
+  }
+
+  distinctVocabulary() {
+    const vocabularyIds = this.props.concepts.map(concept => concept.vocabularyId);
+    return fp.uniq(vocabularyIds);
+  }
+
+
+  filterByVocabulary(vocabulary) {
+    const selectedVocabularies =
+        toggleIncludes(vocabulary, this.state.selectedVocabularies) as unknown as string[];
+    this.filterImageSrc = selectedVocabularies.length > 0 ? 'filtered' : 'filter';
+    this.dt.filter(selectedVocabularies, 'vocabularyId', 'in');
+    this.setState({selectedVocabularies: selectedVocabularies});
+  }
+
+  componentWillReceiveProps(nextProps) {
+    // The purpose of this is to reset the filter on vocabulary on change of domain/concepts
+    if (nextProps.concepts !==  this.props.concepts) {
+      if (this.state.selectedVocabularies) {
+        this.dt.filter([], 'vocabularyId', 'in');
+        this.setState({selectedVocabularies : []});
+      }
+    }
+  }
+
+  render() {
+    const vocabularyFilter = <PopupTrigger
+        side='bottom'
+        content={
+          this.distinctVocabulary().map((vocabulary, i) => {
+            return <div key={i}>
+              <CheckBox style={{marginLeft: '0.2rem', marginRight: '0.3rem'}}
+                        checked={fp.includes(vocabulary, this.state.selectedVocabularies)}
+                        onChange={(checked) => this.filterByVocabulary(vocabulary)}>
+              </CheckBox>
+              <label style={{marginRight: '0.2rem'}}>{vocabulary}</label></div>;
+          })}>
+      <Clickable
+          data-test-id='workspace-menu-button'
+          hover={{opacity: 1}}>
+        <img style={{width: '15%', marginLeft: '-2.5rem'}}
+             src={'/assets/icons/' + this.filterImageSrc + '.svg'}/>
+      </Clickable>
+    </PopupTrigger>;
+    return <div data-test-id='conceptTable'>
+      <DataTable ref={(el) => this.dt = el} value={this.props.concepts}
+                 paginator={true} rows={50} scrollable={true} loading={this.props.loading}
+                 selection={this.state.selectedConcepts}
+                 onSelectionChange={e => this.updateSelectedConceptList(e.value)} >
+      <Column bodyStyle={{...styles.colStyle, width: '3rem'}} headerStyle = {{width: '3rem'}}
```

Comment:
do these columns auto-scale on page-size change?  will coding in a width here make for ugly resizes?